song-server/src/main/java/bio/overture/song/server/service/AnalysisTypeService.java

```diff
@@ -253,9 +253,14 @@ private AnalysisType commitAnalysisType(
     analysisSchema.setVersion(version);
     log.debug("Registered analysisType '{}' with version '{}'", analysisTypeName, version);
     val resolvedSchemaJson = resolveSchemaJsonView(analysisSchema.getSchema(), false, false);
+
+    val resolvedAnalysisSchema = analysisSchemaRepository.findByNameAndVersion(analysisTypeName, version);
+    val createdAt = resolvedAnalysisSchema.isPresent() ? resolvedAnalysisSchema.get().getCreatedAt() : null;
```

Comment:
Replace this with
```java
    val createdAt = analysisSchemaRepository
        .findByNameAndVersion(analysisTypeName, version)
        .map(AnalysisSchema::getCreatedAt)
        .orElseThrow(() -> buildServerException(getClass(), ServerErrors.UNKNOWN_ERROR,
            "Cannot find analysisType with name=%s and version=%s", 
             analysisTypeName, version));
```

Where `UNKNOWN_ERROR` maps to a 500 status code, since this is not the client fault.

Reply:
This is way better, thanks.